src/Model/ApiContext.php

```diff
@@ -0,0 +1,59 @@
+<?php declare(strict_types=1);
+
+namespace SupportPal\ApiClient\Model;
+
+use function sprintf;
+use function trim;
+
+class ApiContext
+{
+    private const BASE_API_PATH = 'api/';
+
+    /** @var string */
+    private $host;
+
+    /** @var string */
+    private $scheme;
+
+    /** @var int */
+    private $port;
+
+    /** @var string */
+    private $path;
+
+    /**
+     * ApiContext constructor.
+     * @param string $host
+     * @param string $scheme
+     * @param int $port
+     * @param string $path
+     */
+    public function __construct(string $host, string $path = '', string $scheme = 'https', int $port = 443)
```

Comment:
I think we could make this fluent and avoid all the optional arguments?
`(new ApiContext($host))->setPath($path)->disableSsl()->setPort($port)`

Reply:
I also wonder whether you could have just said, "give me a url" and then used `parse_url` to extract all these parameters. That may be more user friendly than splitting up the request URI into these pieces

Reply:
That would work, but I feel that enabling the user to pass them is more explicit.

I guess it would be better to limit the constructor to host, and build the others as suggested. I think that's the best approach between both